PWGHF/Utils/utilsMcMatching.h

```diff
@@ -0,0 +1,244 @@
+  {DecayChannelResonant::DstarToDplusToRho0Pi, {+kRho770_0, +kPiPlus}},
+  {DecayChannelResonant::DstarToDplusToF2_1270Pi, {+225, +kPiPlus}},
+};
+
+static const std::unordered_map<DecayChannelMain, const std::vector<int>> DaughtersDstarMain{
+  {DecayChannelMain::DstarToPiKPi, {+kKMinus, +kPiPlus, +kPiPlus}},
+  {DecayChannelMain::DstarToPiKPiPi0, {+kKMinus, +kPiPlus, +kPiPlus, +kPi0}},
+  {DecayChannelMain::DstarToPiKPiPi0Pi0, {+kKMinus, +kPiPlus, +kPiPlus, +kPi0, +kPi0}},
+  {DecayChannelMain::DstarToPiKK, {+kKMinus, +kKPlus, +kPiPlus}},
+  {DecayChannelMain::DstarToPiKKPi0, {+kKMinus, +kKPlus, +kPiPlus, +kPi0}},
+  {DecayChannelMain::DstarToPiPiPi, {+kPiMinus, +kPiPlus, +kPiPlus}},
+  {DecayChannelMain::DstarToPiPiPiPi0, {+kPiMinus, +kPiPlus, +kPiPlus, +kPi0}},
+};
+
+// Lc → p K∓ π±
+static const std::unordered_map<DecayChannelResonant, const std::array<int, 2>> DaughtersLcResonant{
+  {DecayChannelResonant::LcToPKstar0, {+o2::constants::physics::kK0Star892, +kProton}},
+  {DecayChannelResonant::LcToDeltaplusplusK, {+2224, +kKMinus}},
+  {DecayChannelResonant::LcToL1520Pi, {+102134, +kPiPlus}},
+};
+
+static const std::unordered_map<DecayChannelMain, const std::vector<int>> DaughtersLcMain{
+  {DecayChannelMain::LcToPKPi, {+kProton, +kKMinus, +kPiPlus}},
+  {DecayChannelMain::LcToPKPiPi0, {+kProton, +kKMinus, +kPiPlus, +kPi0}},
+  {DecayChannelMain::LcToPPiPi, {+kProton, +kPiMinus, +kPiPlus}},
+  {DecayChannelMain::LcToPKK, {+kProton, +kKMinus, +kKPlus}}};
+
+// Xic → p K∓ π±
+static const std::unordered_map<DecayChannelResonant, const std::array<int, 2>> DaughtersXiCResonant{
+  {DecayChannelResonant::XicToPKstar0, {-o2::constants::physics::kK0Star892, +kProton}},
+  {DecayChannelResonant::XicToPPhi, {+kProton, +o2::constants::physics::kPhi}},
+};
+
+static const std::unordered_map<DecayChannelMain, const std::vector<int>> DaughtersXiCMain{
+  {DecayChannelMain::XicToPKPi, {+kProton, +kKMinus, +kPiPlus}},
+  {DecayChannelMain::XicToPKK, {+kProton, +kKMinus, +kKPlus}},
+  {DecayChannelMain::XicToSPiPi, {+kSigmaPlus, +kPiMinus, +kPiPlus}},
+};
+
+/// Returns a map of the possible final states for a specific 3-prong particle specie
+/// \param pdgMother PDG code of the mother particle
+/// \return a map of final states with their corresponding PDG codes
+inline std::unordered_map<DecayChannelMain, const std::vector<int>> getDecayChannel3Prong(int pdgMother)
+{
+  switch (pdgMother) {
+    case o2::constants::physics::Pdg::kDPlus:
+      return DaughtersDPlusMain;
+    case o2::constants::physics::Pdg::kDS:
+      return DaughtersDsMain;
+    case o2::constants::physics::Pdg::kDStar:
+      return DaughtersDstarMain;
+    case o2::constants::physics::Pdg::kLambdaCPlus:
+      return DaughtersLcMain;
+    case o2::constants::physics::Pdg::kXiCPlus:
+      return DaughtersXiCMain;
+    default:
+      LOG(error) << "Unknown PDG code for 3-prong final states: " << pdgMother;
```

Comment:
I don't see the point of continuing if this happens. I would throw a fatal here.